src/components/icon/icon.ts

```diff
@@ -27,9 +28,18 @@ const iconCache = new Map<string, Promise<SVGResult>>();
 export default class SlIcon extends ShoelaceElement {
   static styles: CSSResultGroup = styles;
 
+  private initialRender = false;
+
   /** Given a URL, this function returns the resulting SVG element or an appropriate error symbol. */
-  private static async resolveIcon(url: string): Promise<SVGResult> {
+  private async resolveIcon(url: string, library?: IconLibrary): Promise<SVGResult> {
     let fileData: Response;
+
+    if (library?.svgSymbolSprite) {
+      return html`<svg @load=${() => this.emit('sl-load')} @error=${() => this.emit('sl-error')} part="base">
```

Comment:
This is fascinating. `load` and `error` events never fire when using spritesheets. I dont know if its an oversight in the spec, but is super interesting.